Emails troubleshooting (#532)
MerginMaps/docs: src/server/troubleshoot/index.md

# Troubleshoot Custom Servers

> This article will help you debug and resolve issues in your <CommunityPlatformNameLink /> or <EnterprisePlatformNameLink /> deployment. If you use the main SaaS <DashboardLink desc="Mergin Maps Server"/>, it is always up-to-date and managed by <MainPlatformName /> team, so report your problems to us as [described here](../../misc/troubleshoot/index.md). Read more about server platforms in [overview article](../index.md). Follow [installation guide](../install/index.md) to install your own server. Documentation to environment variables and other configuration options can be found in [configure](../administer/environment.md) section.

Reviewer: I suggest to change 'SaaS' to 'Cloud'

   
2. Restart the container with the `MERGIN_BASE_URL` variable

## Emails are not sent

> If you are not receiving emails, check that the following environment variables are set correctly:

Reviewer 2: ```suggestion
If you are not receiving emails, check that the following [environment variables](../administer/environment/) are set correctly:
```

> * `MAIL_DEFAULT_SENDER` is mandatory and should be a valid email address

Reviewer 2: ```suggestion
* `MAIL_DEFAULT_SENDER` needs to be a valid email address
```

> * `MAIL_SERVER` should be a valid url to SMTP server

Reviewer 2: ```suggestion
* `MAIL_SERVER` should be a valid URL to SMTP server
```

> * `MAIL_PORT` if you have different port than 587, configure it with this variable

Reviewer 2: ```suggestion
* `MAIL_PORT` SMTP mail server port is set to `587` by default. Change it to the correct value, if you use a different port.
```

> * `MAIL_SUPPRESS_SEND` check if you set it to `False`

Reviewer 2: ```suggestion
* `MAIL_SUPPRESS_SEND` should be set to `false`
```

> In some deployments are SMTP servers that are not supporting authentication and TLS. In this case, you can disable authentication by not defining variable `MAIL_USERNAME` and `MAIL_PASSWORD` (which are by default `None`). If your SMTP server does not support TLS or SSL you can disable encryption by setting `MAIL_USE_TLS` and `MAIL_USE_SSL` to `false`. It's recommended to use authentication and TLS encryption.

Reviewer 2: ```suggestion
In some deployments, there may be SMTP servers that do not support authentication and TLS. In that case, you can disable authentication by not defining variables `MAIL_USERNAME` and `MAIL_PASSWORD` (which are set to `None` by default). 

If your SMTP server does not support TLS or SSL, you can disable encryption by setting `MAIL_USE_TLS` and `MAIL_USE_SSL` to `false`. However, it is recommended to use authentication and TLS encryption.
```

> Server is sending emails with celery worker. If you are not receiving emails, check if celery worker is running. You can check logs for sending emails in `celery-worker` container:

Reviewer 2: ```suggestion
Server is sending emails with a celery worker. If you are not receiving emails, check if celery worker is running. Check logs for sending emails in the `celery-worker` container:
```

Reviewer: How about "Emails are sent by celery worker container (link to docker-compose file)."

```shell

> $ docker logs celery-worker

Reviewer: Shall we add some example of what admins should expect to see?

   
2. Restart the container with the `MERGIN_BASE_URL` variable

## Emails are not sent

Mergin Maps server is sending emails to users in following cases:

* Email verification
* Password reset
* Incoming access request
* Invitation to workspace (only for <EnterprisePlatformName />)

> 

Reviewer 3: I'd prefer not mentioning the exact situations when we send emails. These can change and even the following list is not complete (it is missing the "account created" email).
